hpx/runtime/threads/coroutines/coroutine_fwd.hpp

```diff
@@ -34,11 +34,6 @@
 
 namespace hpx { namespace threads
 {
-    class HPX_EXPORT thread_data;
```

Comment:
The `HPX_EXPORT` needs to go somewhere else, now, I think.

Reply:
it's in thread_id_type.hpp now.

Reply:
So do other files now.

Reply:
You are saying that this PR should fix all includes to not rely on thread_id_type being implicitly defined by some other include?

Reply:
Not necessarily. OTOH, we're trying to cleanup our header-include situation. So when introducing a new header file, perhaps it might make sense to introduce it properly... We could even add an inspect check enforcing this.

Reply:
Currently, the header exposing everything was thread_data_fwd.hpp, which still holds. Depending on how fine-grained we want to resolve those headers...
I'd do this in a separate PR though to not conflate those more or less unrelated changes with the header fixups.

Reply:
ok